ZydisEncoder/generate_encoder_types.py

```diff
@@ -0,0 +1,255 @@
+            #define strncpy(...)
+            #define strpbrk(...)
+            #define strrchr(...)
+            #define strspn(...)
+            #define strstr(...)
+            #define strtok(...)
+            #define strxfrm(...)
+        """,
+        'stdint.h': """
+            typedef signed char        int8_t;
+            typedef short              int16_t;
+            typedef int                int32_t;
+            typedef long long          int64_t;
+            typedef unsigned char      uint8_t;
+            typedef unsigned short     uint16_t;
+            typedef unsigned int       uint32_t;
+            typedef unsigned long long uint64_t;
+            typedef unsigned int       size_t;
+            typedef int                ptrdiff_t;
+            typedef int                intptr_t;
+            typedef unsigned int       uintptr_t;
+
+            #define INT8_MIN           (-127 - 1)
+            #define INT16_MIN          (-32767 - 1)
+            #define INT32_MIN          (-2147483647 - 1)
+            #define INT64_MIN          (-9223372036854775807 - 1)
+            #define INT8_MAX           127
+            #define INT16_MAX          32767
+            #define INT32_MAX          2147483647
+            #define INT64_MAX          9223372036854775807
+            #define UINT8_MAX          0xff
+            #define UINT16_MAX         0xffff
+            #define UINT32_MAX         0xffffffff
+            #define UINT64_MAX         0xffffffffffffffff""",
+    }
+
+    def __init__(self, zydis_path):
+        super().__init__()
+        self.zydis_path = zydis_path
+        self.add_path(os.path.join(zydis_path, 'dependencies', 'zycore', 'include'))
+        self.add_path(os.path.join(zydis_path, 'include'))
+        self.line_directive = None
+        self.define('__x86_64__')
+        with open(os.path.join(zydis_path, 'src', 'Encoder.c')) as f:
+            self.parse(f.read())
+
+    def on_file_open(self, is_system_include, includepath):
+        if is_system_include:
+            filename = os.path.basename(includepath)
+            if filename in ZydisPreprocessor.FAKE_INCLUDES:
+                return io.StringIO(ZydisPreprocessor.FAKE_INCLUDES[filename])
+            elif filename == 'ZydisExportConfig.h':
+                return open(os.path.join(self.zydis_path, 'assets', 'ZydisExportConfigSample.h'))
+
+        return super().on_file_open(is_system_include, includepath)
+
+    def get_preprocessed_file(self):
+        file = io.StringIO()
+        self.write(file)
+        file.seek(0)
+        return file.read()
+
+    @staticmethod
+    def concatenate_tokens(tokens):
+        return ''.join([token.value for token in tokens])
+
+    def expand_macro(self, macro):
+        if isinstance(macro, str):
+            macro = self.macros[macro].value
+        else:
+            macro = macro.value
+        return self.concatenate_tokens(self.expand_macros(macro))
+
+
+class ZydisParser(object):
```

Comment:
```suggestion
class ZydisParser:
```

No longer required in Python 3, inheritance from `object` is now automatic.